// bootstrap-vue-next/dist/src/components/index.mjs
import { _, a } from "../../BAccordionItem.vue_vue_type_script_setup_true_lang-CpORCzCe.mjs";
import { B } from "../../BAlert-BNMW3CdV.mjs";
import { _ as _2, a as a2 } from "../../BAvatarGroup.vue_vue_type_script_setup_true_lang-DcIaqT4o.mjs";
import { _ as _3 } from "../../BBadge.vue_vue_type_script_setup_true_lang-Ba4qWgRD.mjs";
import { _ as _4, a as a3 } from "../../BBreadcrumb.vue_vue_type_script_setup_true_lang-BBFVp3_7.mjs";
import { _ as _5 } from "../../BButton.vue_vue_type_script_setup_true_lang-D7bZQchs.mjs";
import { _ as _6, a as a4 } from "../../BButtonToolbar.vue_vue_type_script_setup_true_lang-CxmPqgRk.mjs";
import { _ as _7 } from "../../BCloseButton.vue_vue_type_script_setup_true_lang-CiWwt6-j.mjs";
import { _ as _8, a as a5, b, c, d, e, f } from "../../BCard.vue_vue_type_script_setup_true_lang-C8uldxyB.mjs";
import { _ as _9, a as a6 } from "../../BCardText.vue_vue_type_script_setup_true_lang-CFyQWiUL.mjs";
import { _ as _10, a as a7 } from "../../BCarouselSlide.vue_vue_type_script_setup_true_lang-DehyGYhk.mjs";
import { _ as _11 } from "../../BCollapse.vue_vue_type_script_setup_true_lang-DigjRrkA.mjs";
import { _ as _12 } from "../../BCol.vue_vue_type_script_setup_true_lang-Dj63q2KG.mjs";
import { _ as _13 } from "../../BContainer.vue_vue_type_script_setup_true_lang-D0WIGZ2B.mjs";
import { _ as _14 } from "../../BRow.vue_vue_type_script_setup_true_lang-CeRjiYD1.mjs";
import { B as B2 } from "../../BDropdown-CE9W2Dxt.mjs";
import { _ as _15, a as a8, b as b2, c as c2, d as d2, e as e2, f as f2 } from "../../BDropdownText.vue_vue_type_script_setup_true_lang-Be9ZyM6V.mjs";
import { _ as _16 } from "../../BForm.vue_vue_type_script_setup_true_lang-Cqtoh3CF.mjs";
import { _ as _17, a as a9 } from "../../BFormFloatingLabel.vue_vue_type_script_setup_true_lang-CVJcElL0.mjs";
import { _ as _18, a as a10, b as b3, c as c3 } from "../../BFormValidFeedback.vue_vue_type_script_setup_true_lang-DuXhe4J8.mjs";
import { _ as _19, a as a11 } from "../../BFormCheckboxGroup.vue_vue_type_script_setup_true_lang-Crv_7H47.mjs";
import { _ as _20 } from "../../BFormFile.vue_vue_type_script_setup_true_lang-6tVyhwhL.mjs";
import { _ as _21 } from "../../BFormGroup.vue_vue_type_script_setup_true_lang-DlTrRdKp.mjs";
import { _ as _22, _ as _23 } from "../../BFormInput.vue_vue_type_script_setup_true_lang-PNXwvQsw.mjs";
import { _ as _24, a as a12 } from "../../BFormRadioGroup.vue_vue_type_script_setup_true_lang-j2Yeg8fd.mjs";
import { _ as _25, a as a13 } from "../../BFormSelect.vue_vue_type_script_setup_true_lang-DW56Y0Hv.mjs";
import { _ as _26 } from "../../useFormSelect-DujwNrDO.mjs";
import { _ as _27 } from "../../BFormSpinbutton.vue_vue_type_script_setup_true_lang-DV-f3D16.mjs";
import { a as a14, _ as _28 } from "../../BFormTags.vue_vue_type_script_setup_true_lang-CwSBK6l0.mjs";
import { _ as _29 } from "../../BFormTextarea.vue_vue_type_script_setup_true_lang-BnsriyGk.mjs";
import { _ as _30 } from "../../BImg.vue_vue_type_script_setup_true_lang-pnma2W3z.mjs";
import { _ as _31, a as a15 } from "../../BInputGroupText.vue_vue_type_script_setup_true_lang-d8oWuUYr.mjs";
import { _ as _32 } from "../../BLink.vue_vue_type_script_setup_true_lang-CsQYhGWY.mjs";
import { _ as _33, a as a16 } from "../../BListGroupItem.vue_vue_type_script_setup_true_lang-DSDkYFie.mjs";
import { B as B3, _ as _34 } from "../../BModalOrchestrator.vue_vue_type_script_setup_true_lang-Bwx2iZbK.mjs";
import { _ as _35, a as a17, b as b4, c as c4, d as d3 } from "../../BNavText.vue_vue_type_script_setup_true_lang-BJ6QFz_P.mjs";
import { _ as _36, a as a18, b as b5, c as c5 } from "../../BNavbarToggle.vue_vue_type_script_setup_true_lang-BpYt-oBq.mjs";
import { B as B4 } from "../../BOffcanvas-DIBIwNtP.mjs";
import { _ as _37 } from "../../BOverlay.vue_vue_type_script_setup_true_lang-C4SWC8DS.mjs";
import { _ as _38 } from "../../BPagination.vue_vue_type_script_setup_true_lang-6OTRvX9H.mjs";
import { _ as _39, a as a19, b as b6, c as c6, d as d4 } from "../../BPlaceholderWrapper.vue_vue_type_script_setup_true_lang-BFsZiIaA.mjs";
import { B as B5 } from "../../floatingUi-D5nPqieE.mjs";
import { _ as _40 } from "../../BPopoverOrchestrator.vue_vue_type_script_setup_true_lang-BhzsrsN3.mjs";
import { _ as _41, a as a20 } from "../../BProgress.vue_vue_type_script_setup_true_lang-CjEfC0Pz.mjs";
import { _ as _42 } from "../../BSpinner.vue_vue_type_script_setup_true_lang-C2afVNWl.mjs";
import { _ as _43, a as a21, b as b7, c as c7, d as d5, e as e3, f as f3, g } from "../../BTable.vue_vue_type_script_setup_true_lang-BhFlxAjx.mjs";
import { _ as _44 } from "../../BTableSimple.vue_vue_type_script_setup_true_lang-DOAS4KSP.mjs";
import { _ as _45, a as a22 } from "../../BTabs.vue_vue_type_script_setup_true_lang-DR3YhpHo.mjs";
import { B as B6, _ as _46 } from "../../BToastOrchestrator.vue_vue_type_style_index_0_lang-DKAZyKKr.mjs";
import { _ as _47 } from "../../BTooltip.vue_vue_type_script_setup_true_lang-C3ZQMvAC.mjs";
export {
  _ as BAccordion,
  a as BAccordionItem,
  B as BAlert,
  _2 as BAvatar,
  a2 as BAvatarGroup,
  _3 as BBadge,
  _4 as BBreadcrumb,
  a3 as BBreadcrumbItem,
  _5 as BButton,
  _6 as BButtonGroup,
  a4 as BButtonToolbar,
  _8 as BCard,
  a5 as BCardBody,
  b as BCardFooter,
  _9 as BCardGroup,
  c as BCardHeader,
  d as BCardImg,
  e as BCardSubtitle,
  a6 as BCardText,
  f as BCardTitle,
  _10 as BCarousel,
  a7 as BCarouselSlide,
  _7 as BCloseButton,
  _12 as BCol,
  _11 as BCollapse,
  _13 as BContainer,
  B2 as BDropdown,
  _15 as BDropdownDivider,
  a8 as BDropdownForm,
  b2 as BDropdownGroup,
  c2 as BDropdownHeader,
  d2 as BDropdownItem,
  e2 as BDropdownItemButton,
  f2 as BDropdownText,
  _16 as BForm,
  _19 as BFormCheckbox,
  a11 as BFormCheckboxGroup,
  _17 as BFormDatalist,
  _20 as BFormFile,
  a9 as BFormFloatingLabel,
  _21 as BFormGroup,
  _22 as BFormInput,
  _18 as BFormInvalidFeedback,
  _24 as BFormRadio,
  a12 as BFormRadioGroup,
  a10 as BFormRow,
  _25 as BFormSelect,
  _26 as BFormSelectOption,
  a13 as BFormSelectOptionGroup,
  _27 as BFormSpinbutton,
  a14 as BFormTag,
  _28 as BFormTags,
  b3 as BFormText,
  _29 as BFormTextarea,
  c3 as BFormValidFeedback,
  _30 as BImg,
  _23 as BInput,
  _31 as BInputGroup,
  a15 as BInputGroupText,
  _32 as BLink,
  _33 as BListGroup,
  a16 as BListGroupItem,
  B3 as BModal,
  _34 as BModalOrchestrator,
  _35 as BNav,
  a17 as BNavForm,
  b4 as BNavItem,
  c4 as BNavItemDropdown,
  d3 as BNavText,
  _36 as BNavbar,
  a18 as BNavbarBrand,
  b5 as BNavbarNav,
  c5 as BNavbarToggle,
  B4 as BOffcanvas,
  _37 as BOverlay,
  _38 as BPagination,
  _39 as BPlaceholder,
  a19 as BPlaceholderButton,
  b6 as BPlaceholderCard,
  c6 as BPlaceholderTable,
  d4 as BPlaceholderWrapper,
  B5 as BPopover,
  _40 as BPopoverOrchestrator,
  _41 as BProgress,
  a20 as BProgressBar,
  _14 as BRow,
  _42 as BSpinner,
  _45 as BTab,
  _43 as BTable,
  a21 as BTableLite,
  _44 as BTableSimple,
  a22 as BTabs,
  b7 as BTbody,
  c7 as BTd,
  d5 as BTfoot,
  e3 as BTh,
  f3 as BThead,
  B6 as BToast,
  _46 as BToastOrchestrator,
  _47 as BTooltip,
  g as BTr
};
